src/OrchardCore.Modules/OrchardCore.Liquid/Filters/ShortCodeFilter.cs

```diff
@@ -18,10 +20,20 @@ public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArgument
 
             var shortcodeService = ((IServiceProvider)services).GetRequiredService<IShortcodeService>();
 
-            // TODO This provides no context to the shortcode service.
-            // It could take a content item as an argument to provide some context.
+            var context = new Context();
 
-            return new StringValue(await shortcodeService.ProcessAsync(input.ToStringValue()));
+            // Retrieve the 'ContentItem' from the ambient liquid scope.
+            var model = ctx.LocalScope.GetValue("Model").ToObjectValue();
+            if (model is Shape shape && shape.Properties.TryGetValue("ContentItem", out var contentItem))
+            {
```

Comment:
yep, interesting approach. Will only work from the context of the liquid filter, not with the razor extension, that does the same thing?

Which was why the TODO suggested it as an argument to the filter, so the razor extension could also take an optional argument.
Unless we can attach to the current rendering scope in the razor filter to provide the same? 

That'd be pretty cool.

Reply:
@deanmarcussen I added exactly the same to the razor extension and tried it